src/Raven.Server/Documents/Handlers/Processors/Revisions/RevisionsHandlerProcessorForGetRevisions.cs

```diff
@@ -0,0 +1,173 @@
+                CheckNotModified(actualEtag);
+
+                long numberOfResults;
+                long totalDocumentsSizeInBytes;
+                var blittable = RequestHandler.GetBoolValueQueryString("blittable", required: false) ?? false;
+                if (blittable)
+                {
+                    var revisionResult = new RevisionsResult()
+                    {
+                        Results = revisions.Select(x => x?.Data).ToArray(),
+                        TotalResults = total
+                    };
+
+                    WriteRevisionsBlittable(context, revisionResult, out numberOfResults, out totalDocumentsSizeInBytes);
+                }
+                else
+                {
+                    var revisionResult = new RevisionsResult<Document>()
+                    {
+                        Results = revisions, 
+                        TotalResults = total
+                    };
+                    (numberOfResults, totalDocumentsSizeInBytes) = await WriteRevisionsJsonAsync(context, metadataOnly, revisionResult, token);
+                }
+
+                //using this function's legacy name GetRevisionByChangeVector
+                AddPagingPerformanceHint(PagingOperationType.Documents, "GetRevisionByChangeVector", HttpContext.Request.QueryString.Value, numberOfResults,
+                    revisions.Count, sw.ElapsedMilliseconds, totalDocumentsSizeInBytes);
+            }
+        }
+
+        protected override async ValueTask GetRevisionsAsync(bool metadataOnly, CancellationToken token)
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var sw = Stopwatch.StartNew();
+
+                var id = RequestHandler.GetQueryStringValueAndAssertIfSingleAndNotEmpty("id");
+                var before = RequestHandler.GetDateTimeQueryString("before", required: false);
+                var start = RequestHandler.GetStart();
+                var pageSize = RequestHandler.GetPageSize();
+
+                Document[] revisions = Array.Empty<Document>();
+                long count = 0;
+                if (before != null)
+                {
+                    var revision = RequestHandler.Database.DocumentsStorage.RevisionsStorage.GetRevisionBefore(context, id, before.Value);
+                    if (revision != null)
+                    {
+                        count = 1;
+                        revisions = new[] {revision};
+                    }
+                }
+                else
+                {
+                    (revisions, count) = RequestHandler.Database.DocumentsStorage.RevisionsStorage.GetRevisions(context, id, start, pageSize);
+                }
+
+                var actualChangeVector = revisions.Length == 0 ? "" : revisions[0].ChangeVector;
+                CheckNotModified(actualChangeVector);
```

Comment:
this needs to brake if not modified. Probably you want to return bool from this method?

Reply:
Check how this is handled in `AbstractStudioCollectionsHandlerProcessorForPreviewCollection`, I would prefer to handle that in a similar way.